pymeasure/display/console.py

```diff
@@ -0,0 +1,276 @@
+        # Handle Ctrl+C nicely
+        signal.signal(signal.SIGINT, lambda sig, _: self.abort())
+
+    def get_filename(self, directory):
+        """ Return filename for logging.
+
+        User can override this method to define their own filename
+        """
+        if self.filename is not None:
+            return os.path.join(directory, self.filename)
+        else:
+            return unique_filename(directory)
+
+    def _update_progress(self, progress):
+        if self.bar:
+            self.bar.update(progress)
+
+    def _update_status(self, status):
+        if self.bar:
+            self.bar.update(status=Procedure.STATUS_STRINGS[status])
+
+    def _update_log(self, record):
+        log.emit(record)
+
+    def _failed(self):
+        self._terminate("Manager's running experiment has failed")
+
+    def _abort_returned(self):
+        self._terminate("Running experiment has returned after an abort")
+
+    def _finish(self):
+        self._terminate("Running experiment has finished", 100.0)
+
+    def _terminate(self, debug_message, update_bar=None):
+        log.debug(debug_message)
+        self._monitor.wait()
+        log.debug("Monitor has cleaned up after the Worker")
+        if self.bar:
+            self.bar.update(update_bar)
+            self.bar.finish()
+        self.quit()
+
+    def abort(self):
+        """ Aborts the currently running Experiment, but raises an exception if
+        there is no running experiment
+        """
+        self._worker.update_status(Procedure.ABORTED)
+        self._worker.stop()
+
+    def exec(self):
+        # Parse command line arguments
+        args = vars(self.parser.parse_args(self.args[1:]))
+        procedure = self.procedure_class()
+
+        self.directory = args['log_directory']
+        self.filename = args['log_file']
+        try:
+            log_level = int(args['log_level'])
+        except ValueError:
+            # Ignore and assume it is a valid level string
+            log_level = args['log_level']
+        self.log_level = log_level
+        log.setLevel(self.log_level)
+        self.log.setLevel(self.log_level)
+
+        if args['sequence_file'] is not None:
+            raise NotImplementedError("Sequencer not yet implemented")
+
+        bar_enabled = not args['no_progressbar']
+
+        # Set procedure parameters
+        parameter_values = {}
+
+        if args['use_log_file'] is not None:
+            # Special case set parameters from log file
+            results = Results.load(args['use_log_file'])
+            for name in results.parameters:
+                parameter_values[name] = results.parameters[name].value
+        else:
+            for name in args:
+                opt_name = name.replace("_", "-")
+                if not (opt_name in self.parser.special_options):
+                    parameter_values[name] = args[name]
+
+        procedure.set_parameters(parameter_values)
+        if (bar_enabled):
```

Comment:
since you use `bar_enabled` only here, there seems no need to define the variable at all and just directly access `not args['no_progressbar']` here

Reply:
OK, done.